packages/lesswrong/server/scripts/backfillParentTags.ts

```diff
@@ -0,0 +1,49 @@
+import TagRels from "../../lib/collections/tagRels/collection";
+import Tags from "../../lib/collections/tags/collection";
+import { Globals, updateMutator } from "../vulcan-lib";
+import { v4 as uuid } from 'uuid';
+import { updatePostDenormalizedTags } from "../tagging/tagCallbacks";
+
+const backfillParentTags = async (childTagSlug, parentTagSlug) => {
+  const parentTag = (await Tags.find({slug: parentTagSlug}).fetch())[0];
+  const childTag = (await Tags.find({slug: childTagSlug}).fetch())[0];
+  const parentTagRelPostIds = (await TagRels.find({tagId: parentTag._id}).fetch()).map(rel => rel.postId);
+
+  const childTagRelPostIds = (await TagRels.find({tagId: childTag._id}).fetch()).filter(rel => !parentTagRelPostIds.includes(rel.postId)).map(rel => rel.postId);
+
+  const parentTagRelIds = childTagRelPostIds.map(_ => uuid({}).slice(0, 17));
+  
+  await updateMutator({
+    collection: Tags,
+    documentId: childTag._id,
+    set: { parentTagId: parentTag._id },
```

Comment:
I think it's probably better not to do this and just rely on setting the parent tags manually (and just backfill tagrels automatically), seeing as there aren't that many tags and it simplifies the automatic part. (and also because we've already added most of the [parent/subtag relationships](https://docs.google.com/spreadsheets/d/1UCUgHvTf0yVoy0l9bEtIe_9XKr0WjrhhADKoomVTSa8/edit#gid=2120438285), so any changes now will just be switching things around)

Reply:
Added here https://github.com/ForumMagnum/ForumMagnum/pull/6604/commits/781d41f38f493c6f69e615f7e5e2c969aaaa7711 👍